src/vanilla/ClientModelExtensions.cs

```diff
@@ -370,7 +370,7 @@ public static string CheckNull(string valueReference, string executionBlock)
 
             if (sb.ToString().Trim().Length > 0)
             {
-                if (type.IsValueType())
+                if (type.IsValueType() && sequence != null && !sequence.IsNullable)

```

Comment:
Let's create a `isNullable` local and set it `if (sequence != null && sequence.ShouldValidateChain())` branch.

Reply:
Don't we want to check `sequence.IsNullable`?

Reply:
Yep, we do

Reply:
` sequence` is always null in this case, so I updated the code and passed the whole [variable ](https://github.com/Azure/autorest.csharp/pull/895/commits/3948893c945f85597fdebe3843d4183ce9360dfd#diff-99c896ba66d1a5f58410c2e2a5a02790)instead of just `valueReference` and it already has `IsNullable` method.